### PR TITLE
Added Container Mocks and improved BlockStates as a whole

### PR DESCRIPTION
**Pull request by TheBusyBiscuit**
# Description
This Pull Request improves upon the current BlockState mocks, I have also added TIleStates, Containers, Chests, Barrels and Ender Chest BlockState mocks and Unit Tests corresponding to the newly added mocks.

Many other methods have been added, Metadata has now been implemented too for various mocks and the Metadata Test has been expanded by a Parameterized Test class which can test any `Metadatatable`-Object.

The version was bumped to 0.5.0.<br>
I have also done some refactoring and improvements on other files, I updated the Maven-portion in the README too and removed the "releases info" at the end since it no longe applies. Sorry for making the diff this huge.

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Version number updated in `settings.gradle`.
- [x] Unit tests added.
- [x] Code follows existing code format.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/ServerMock.java [hunk 1]
```diff
@@ -1453,10 +1453,24 @@ public int getTicksPerWaterAmbientSpawns()
...
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		assertMainThread();
+
+		Iterator<Recipe> iterator = recipeIterator();
+
+		while (iterator.hasNext())
+		{
+			Recipe recipe = iterator.next();
```
Comment (seeseemelk) [inline src/main/java/be/seeseemelk/mockbukkit/ServerMock.java]: Can't this be simplified with a `for (Recipe recipe : recipeIterator())`?

**Review by seeseemelk — APPROVED** [review]
Looks good except for maybe one small change, although even without that it's good enough for me.

File: src/main/java/be/seeseemelk/mockbukkit/ServerMock.java (hunk 1 above)
Comment (TheBusyBiscuit) [reply src/main/java/be/seeseemelk/mockbukkit/ServerMock.java]: Oh yes, of course!
I'll have to admit that I literally just copied #removeRecipe and didn't really put much thought into it. 
I replaced it with a simple for-loop now!

**Review by seeseemelk — APPROVED** [review]

**Comment by seeseemelk** [comment]
MockBukkit 0.5.0 has been released

**Pull request merged**